packages/react-sdk/src/components/FallbackSnapshotProvider/FallbackSnapshotProvider.tsx

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2024 Nordeck IT + Consulting GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  RoomMemberStateEventContent,
+  StateEvent,
+} from '@matrix-widget-toolkit/api';
+import { useWidgetApi } from '@matrix-widget-toolkit/react';
+import { PropsWithChildren, useEffect } from 'react';
+import { RoomEncryptionEvent, RoomHistoryVisibilityEvent } from '../../model';
+import { useActiveWhiteboardInstance } from '../../state';
+import {
+  useGetRoomEncryptionQuery,
+  useGetRoomHistoryVisibilityQuery,
+  useGetRoomMembersQuery,
+} from '../../store/api';
+
+export function FallbackSnapshotProvider({ children }: PropsWithChildren<{}>) {
+  const whiteboardInstance = useActiveWhiteboardInstance();
+  const { data: roomMembers } = useGetRoomMembersQuery();
+  const { data: roomEncryption } = useGetRoomEncryptionQuery();
+  const { data: roomHistoryVisibility } = useGetRoomHistoryVisibilityQuery();
+  const ownUserId = useWidgetApi().widgetParameters.userId;
+
+  useEffect(() => {
```

Comment:
On load, do we already know the entire state? Means do we know the snapshots and invites? If not, we should delay the logic to give NeoBoard some time to retrieve all the data.